shadow-dom/events/event-dispatch/test-003.html

```diff
@@ -37,19 +37,29 @@
     var s = host.createShadowRoot();
 
     var div1 = d.createElement('div');
-    div1.setAttribute('style', 'height:100%; width:100%');
+    div1.setAttribute('style', 'height:50%; width:100%');
     div1.setAttribute('id', 'div1');
     s.appendChild(div1);
 
-    s.addEventListener('mouseover', A_05_05_03_T01.step_func(function(event) {
+    var div2 = d.createElement('div');
+    div2.setAttribute('style', 'height:50%; width:100%');
+    div2.setAttribute('id', 'div2');
+    s.appendChild(div2);
+
+    // Add an event listner above its shadow root. If both target and
+    // relatedTarget are in the same shadow tree under the root,
+    // retargeted target and relatedTarget will be the same node
+    // and the event must not invoke the handler.
+    host.addEventListener('mouseover', A_05_05_03_T01.step_func(function(event) {
```

Comment:
IIUC a trusted event must not invoke an event, if target and relatedTarget are same.
This test dispatches a "mouseover" event whose target is div1 and relatedTarget is div2,
and div1 and div2 are in the same TreeScope.  So once the event bubbles up out of the
TreeScope, target and relatedTarget become same and will not invoke the event handler.

Or are you saying we should dispatch "trusted event" for this test?
I don't think we have interoperable way to dispatch a trusted event in web platform tests.